docs/development.txt

```diff
@@ -0,0 +1,82 @@
+
+
+[[development-decisions]]
+== Development Decisions
+
+Many development decisions will be made during the day-to-day work of JanusGraph contributors without any extra process overhead.
+However, for larger bodies of work and significant updates and additions, the following process shall be followed.
+Significant work may include, but is not limited to:
+
+* A major new feature or subproject, e.g., a new storage adapter
+* Incrementing the version of a core dependency such as a Apache TinkerPop
+* Addition or deprecation of a public API
+* Internal shared data structure or API change
+* Addition of a new major dependency
+
+For these sorts of changes, contributors will:
+
+* Create one or more issues in the GitHub issue tracker
+* Start a DISCUSS thread on the https://groups.google.com/forum/#!forum/janusgraph-dev[janusgraph-dev] list where the proposed change may be discussed by committers
+and other community members
+* When the proposer feels it appropriate, a VOTE shall be called
+* Two +1 votes are required for the change to be accepted
+
+[[branching]]
+== Branching
+
+For features that involve only one developer, developers will work in their own JanusGraph forks, managing their own branches, and submitting pull requests when ready. If multiple developers wish to collaborate on a feature,
+they may request that a feature branch be created in JanusGraph repository.  If the developers are not committers, a JanusGraph committer will be assigned as the shepherd for that branch.  The shepherd will be responsible for merging pull requests to that branch.
+
+=== Branch Naming Conventions
+
+All branch names shall be prepended with `Issue_#_`.
+
+[[pull-requests]]
+== Pull Requests
+
+Most changes to JanusGraph must be reviewed before they are committed (RTC - review-then-commit).
+This workflow is performed using GitHub's pull request process. In instances where the committer deems the full process unnecessary, commit-then-review (CTR) may be employed.
+This should be used infrequently and only for extremely trivial changes.
+Changes of this sort may include:
+
+* Documentation typo or small documentation addition
+* Addition of a new test case
+
+Any commit that is made following CTR shall include the fact that it is a CTR in the commit comments.
+The committer should also send a message to https://groups.google.com/forum/#!forum/janusgraph-dev[janusgraph-dev] to notify other project members of the CTR.
+If another committer responds with a -1, the commit should be rolled back and the formal RTC process should be followed.
+
+Users wishing to contribute to JanusGraph should fork the https://github.com/janusgraph/janusgraph[JanusGraph] repository and then submit pull requests using the GitHub pull request process.
+Every pull request must be associated with an existing issue.
+Be sure to include the relevant issue number in the title of your pull request and a description of what test suites were run to validate the pull request.
```

Comment:
CI will run tests for you (ideally). Eventually, we should probably ask for contributors to note any special testing steps they took.